project/ws/app/src/lib/routes/gyaan-karmayogi/components/gyaan-karmayogi-view-all/gyaan-karmayogi-view-all.component.ts

```diff
@@ -193,4 +198,79 @@ seeAllPageConfig: any
     }
     return v6filters
   }
+
+  // the bellow method is used to get initial facet data to filters
+  getFacetsData() {
+    this.filterDataLoading = true
+    const request = this.route.snapshot.data.pageData &&
+    this.route.snapshot.data.pageData.data &&
+    this.route.snapshot.data.pageData.data.facetReqData &&
+    this.route.snapshot.data.pageData.data.facetReqData
+    this.seeAllSvc.searchV6(request).subscribe((response: any) => {
+      if (response &&  response.result &&
+         response.result.facets) {
+          const localFacetData: any = {
+            sectorName: {
+              name: 'Sector',
+              values: 'values',
+            },
+            subSectorName: {
+              name: 'Sub sector',
+              values: 'values',
+            },
+            // resourceCategory: {
+            //   name: 'Category',
+            //   values: 'values',
+            // },
+          }
+          response.result.facets.forEach((facet: any) => {
+            if (localFacetData[facet.name]) {
+              facet.values.forEach((item: any) => {
+                if (item.name === this.keyData.toLowerCase()) {
+                  item['checked'] = true
+                  this.selectedFilter['resourceCategory'] = [item.name]
+                }
+              })
+              localFacetData[facet.name].values = facet.values
+            }
+          })
+
+          this.facetsData = localFacetData
+          this.facetsDataCopy = localFacetData
+
+         }
+         this.filterDataLoading = false
+    },                                         (_error: any) => {
+
+    })
+  }
+
+  // the below method is used to get emitted value from filter component
+  filterChange(event: any) {
+    this.changeSelection(event.event, event.key, event.keyData)
+  }
+
+  // the below method used to form the filters and call api
+  changeSelection(event: any, key: any, keyData: any) {
+    keyData['checked'] = event
+    if (this.selectedFilter[key] && this.selectedFilter[key].includes(keyData.name)) {
```

Comment:
Check for this.selectedFilter as well , because if key not available it will throw an error